### DISCUSSION TITLE
Help setting IMSAI 8080 IP address to static

### DISCUSSION THREAD

**Discussion by jasonjdull** [thehighnibble/thehighnibble.github.io#153] [IMSAI8080esp replica]
Hello, I'm trying to set a static IP on my IMSAI 8080 replica. I am able to connect to it on my wifi with automatic IP selection, but I haven't been able to find anything so far to get it to set a static IP. 
I tried various ways in the boot.conf file such as adding:

set_ipv4_address=192.168.2.114
set_ipv4_gateway=192.168.2.1
set_ipv4_netmask=255.255.255.0

or

sta_if.active=True
sta_if.ifconfig='192.168.2.114', '255.255.255.0', '192.168.2.1', '8.8.8.8'

or

BOOTPROTO=static
DNS1=8.8.8.8
DNS2=4.4.4.4
GATEWAY=192.168.2.1
IPADDR=192.68.2.114/24

Any help is appreciated! Thanks,

Jason

**Comment by udo-munk** [comment]
A machine on a Wifi cannot just set it's wanted IP address static, if two ore more would try the same, this would result in chaos. So there is a management instance, that makes sure that each machine gets a unique IP, this is a little DNS in your Wifi router. Usually this has a configuration dialog, where you assign an IP to the hardware address of a machine. With doing this the DNS will then always assign this IP address, so that it is quasi static.

**Reply by rsiggy** [reply]
The configuration udo-munk mentions is usually in the DHCP configuration section of your AP/Router.  Look for DHCP address reservation - it will probably be under one of the "Advanced" configuration tabs of most home routers.  Typically the configuration will scan for connected wifi devices and present their MAC addresses and maybe the device name if it has been so configured.  You then match the MAC address from the IMSAI to the desired IP address and you'll be all set.

**Reply by david-taylor** [reply]
I tried that but if I leave the Imsai 8080 off for a week, the MAC address changes rendering any MAC to IP mapping useless. Is there a way to lock the MAC or IP address in the boot.conf?

**Comment by udo-munk** [comment]
The ESP32 WiFi MAC address is hard-coded in hardware by the manufacturer and won't change from allone.

**Reply by david-taylor** [reply]
My error. Thanks for the clarification.

**Comment by MarkusPlehn** [comment]
i fixed my IMSAI8080 IP via router (Speedport3): 
set the router DHCP-range from ...110 to ...111 , delete all DHCP saved MAC-adrresses in router, switched off all other WiFi devices und start the IMSAI8080. Then it will get the IP ...110 from router. Now: "allways assign this IP to device (MACAdress: IMSAI8080)" or what dialog in your router will shown .
From now allways IP 192.168.178.110

So: what Udo Munk said :thumbsup:​​​​

**Comment by udo-munk** [comment]
Excellent.
As far as I know every WiFi router allows to assign an IP to the MAC address of a device, so that it will become static. The exact procedure will be different and there are many WiFi routers with different configuration dialogs, so we can only try to explain the basics how to do this.

**Comment by mgthomas19422** [comment]
My firewall router DHCP service does not allow fixed MAC->IP DHCP assignments, and my WiFi does support configuring some WiFi clients with static IPs configured on those devices, so long as I pick IPs outside the DHCP range to avoid conflicts. I have other wireless devices where I am using configuration options within those devices to disable "DHCP" and set their static wireless IPs.

The original question of whether and how it is possible to assign a static IP address via boot.conf remains unanswered. Is there a list of boot.conf options, beyond those in the provided sample file?
